[D2] Button Component UI 구현

kimkyunghun3 opened the pull request:
## 작업 내용
Button Component UI 구현
(아래 스크린샷 부분 구현)

### 스크린샷
<img width="422" alt="스크린샷 2022-06-26 오전 3 32 53" src="https://user-images.githubusercontent.com/52434820/175786247-2e963469-ac4f-4af5-8b37-1d68364a2e78.png">

kimkyunghun3 committed:
:bulb: coButtonStyle 구현

# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

################
# 세부 내용을 아랫줄에 작성

################
# 추가적인 세부 내용을 아랫줄에 작성

################
# :tada: = 프로젝트 시작
# :sparkles: = create 파일 생성 시
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시
# :rocket: = release 배포 시
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우)
# :memo: = 문서 작성 시
# :art: = 디자인 시스템 관련 내용 (Color / Asset)
# :hammer: = 테스트 코드 작성
################

kimkyunghun3 committed:
:bulb: commonButton 구현

# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

################
# 세부 내용을 아랫줄에 작성

################
# 추가적인 세부 내용을 아랫줄에 작성

################
# :tada: = 프로젝트 시작
# :sparkles: = create 파일 생성 시
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시
# :rocket: = release 배포 시
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우)
# :memo: = 문서 작성 시
# :art: = 디자인 시스템 관련 내용 (Color / Asset)
# :hammer: = 테스트 코드 작성
################

AhnSangHee commented on Projects/DesignSystem/Sources/Component/Button/CoButtonStyle.swift:
```diff
@@ -0,0 +1,52 @@
+
+public struct CoButtonStyle {
+    
+    public enum buttonHeight: CGFloat {
```
```suggestion
    public enum ButtonHeight: CGFloat {
```
🟡 enum 타입 네이밍 수정해주시면 좋을 것 같아요!

kimkyunghun3 replied:
앗 감사합니다! 수정했습니다!

szzang-kr commented on Projects/DesignSystem/Sources/Component/Button/CommonButton.swift:
```diff
@@ -0,0 +1,34 @@
+    }
+    
+    public func setUI() {
+        self.layer.cornerRadius = self.style.radius
+        self.setTitleColor(self.style.titleColor, for: .normal)
+        self.backgroundColor = self.style.buttonColor
+        self.titleLabel?.font = self.style.buttonFont
+        self.setTitle(self.style.title, for: .normal)
+        self.contentEdgeInsets = UIEdgeInsets(top: self.style.topInset, left: self.style.leftInset, bottom: self.style.bottomInset, right: self.style.rightInset)
+        self.titleLabel?.pin.height(self.style.height.rawValue)
```
❓ `self.~ = self.~` 로 모든 코드가 작성되어있는데 왜 self를 붙여야 하나요?

kimkyunghun3 replied:
self를 명시하지 않더라도 파악할 수 있기 때문에 생략할 수 있습니다. 이는 제 생각에는 취향차이(?)라고 생각해서 이를 통일성 있게 작성해주면 괜찮다고 생각했습니다

szzang-kr commented on Projects/DesignSystem/Sources/Component/Button/CommonButton.swift:
```diff
@@ -0,0 +1,34 @@
+import UIKit
+import PinLayout
+
+public class CommonButton: UIButton {
```
❓ CoButtonStyle과 CommonButton의 네이밍은 어떤 관계가 있나요?

kimkyunghun3 replied:
네이밍 부분에서는 아직 앱 이름이 확정이 안되어서 임시로 적었는데 수정하지 않았네요. 앱이름이 확정된 상태이므로 DDIPButton, DDIPButtonStyle로 통일되게 수정했습니다

szzang-kr reviewed (commented):
해커톤 하느라 고생많았어요 😃

AhnSangHee commented on Projects/DesignSystem/Sources/Component/Button/CoButtonStyle.swift:
```diff
@@ -0,0 +1,52 @@
+
+import UIKit
+
+public struct CoButtonStyle {
```
❓ CoButton에서 Co가 무슨 줄임말인지 여쭤봐도 될까요?!

kimkyunghun3 replied:
common을 줄여서 쓰려고 했는데 정확하게 적지 않았네요! 앱이름 나와서 DDIPButtonStyle로 통일되게 네이밍 바꾸었습니다. 감사합니다

AhnSangHee reviewed (approved):
1차 해커톤 수고 많으셨습니다아🙂🙂🙂

kimkyunghun3 committed:
:recycle: Component Button 관련 네이밍 수정 #5

# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

################
# 세부 내용을 아랫줄에 작성

################
# 추가적인 세부 내용을 아랫줄에 작성

################
# :tada: = 프로젝트 시작
# :sparkles: = create 파일 생성 시
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시
# :rocket: = release 배포 시
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우)
# :memo: = 문서 작성 시
# :art: = 디자인 시스템 관련 내용 (Color / Asset)
# :hammer: = 테스트 코드 작성
################
